torch/_dynamo/test_case.py

```diff
@@ -102,16 +102,25 @@ def tearDown(self) -> None:
             torch.set_grad_enabled(self._prior_is_grad_enabled)
 
     def assertEqual(self, x: Any, y: Any, *args: Any, **kwargs: Any) -> None:  # type: ignore[override]
-        if (
-            config.debug_disable_compile_counter
-            and isinstance(x, utils.CompileCounterInt)
-            or isinstance(y, utils.CompileCounterInt)
-        ):
-            return
+        if config.debug_disable_compile_counter:
+            if isinstance(x, utils.CompileCounterInt) or isinstance(
+                y, utils.CompileCounterInt
+            ):
+                return
+            # skip checks like self.assertEqual(len(counters["graph_break"]), 1)
+            if (
+                (cur_frame := inspect.currentframe())
+                and (upper_frame := cur_frame.f_back)
+                and (upper_code := inspect.getframeinfo(upper_frame).code_context)
```

Comment:
Quick question: Does this include the case when there are 3 layer of nested function?
``f(g(h(x)))``

Reply:
The current frame would be `self.assertEqual` and the second frame would be the test file - what other case are you thinking about?

This is also admittedly hacky - I'll probably remove this once `nested_graph_breaks` is enabled on dynamo unittests.